Feature/reservation

itzel-aguirre opened the pull request:
Genere 4 archivos:

- GenerousBO.php
- Generous.php
- reservations.php
- reservationsBO.php

Se encuentran las funciones para Insertar, Actualizar y eliminar

itzel-aguirre committed:
DML model

itzel-aguirre committed:
Merge branch 'master' of github.com:royal-woods/jazz into feature/reservation

amilkhael commented:
En el título del Pull va lo que hiciste, no se por ejemplo crear clases para la gestión
 reservaciones

amilkhael commented on model/Generous.php:
```diff
@@ -0,0 +1,34 @@
+<?php
+class GenerousBO
```
```suggestion
class GenreBO
```

amilkhael commented on model/Generous.php:
```diff
@@ -0,0 +1,34 @@
+class GenerousBO
+{
+//Attributes
+private $id_gender;
```
```suggestion
private $id_genre;
```

amilkhael commented on model/Generous.php:
```diff
@@ -0,0 +1,34 @@
+{
+//Attributes
+private $id_gender;
+private $gender;
```
```suggestion
private $genre;
```

amilkhael commented on model/GenerousBO.php:
```diff
@@ -0,0 +1,64 @@
+include_once 'ConectaBD.php';
+include_once 'Generous.php';
+
+class reservationsBO
```
```suggestion
class ReservationsBO
```

amilkhael commented on model/GenerousBO.php:
```diff
@@ -0,0 +1,64 @@
+  {
+    $databaseConected = new ConectDB();
+    $databaseConected->conectar();
+    $query = "DELETE FROM `generos` WHERE `id_genero`=".$generousDData->gender." ";
```
$generousDData no corresponde con lo que estas recibiendo en la función, mañana te enseño a probarlo

amilkhael commented on model/GenerousBO.php:
```diff
@@ -0,0 +1,64 @@
+    }
+    else{
+        $databaseConected->desconectar();
+        return "Error: " . mysqli_error($databaseConected);
```
Debes regresar el error como json

amilkhael commented on model/GenerousBO.php:
```diff
@@ -0,0 +1,64 @@
+    $databaseConected->conectar();
+    $query = "DELETE FROM `generos` WHERE `id_genero`=".$generousDData->gender." ";
+    $databaseConected->consulta($query);
+    if ($databaseConected->query($query) == FALSE) {
```
Verifica bien que te devuelve el delet creo un True cuando todo fue exitoso

amilkhael commented on model/GenerousBO.php:
```diff
@@ -0,0 +1,64 @@
+   
+    if ($ReservationInfo->num_rows > 0) {
+        $databaseConected->desconectar();
+        return json_encode (array('error'=>'Rerervación registrada'));
```
Esto no es un error

amilkhael commented on model/reservationsBO.php:
```diff
@@ -0,0 +1,98 @@
+
+    if ($databaseConected->query($query) == FALSE) {
+        $databaseConected->desconectar();
+        return "Genero Eliminado";
```
Regresalo como Json

amilkhael commented on model/reservationsBO.php:
```diff
@@ -0,0 +1,98 @@
+    }
+    else{
+        $databaseConected->desconectar();
+        return "Error: " . mysqli_error($databaseConected);
```
Regresalo como JSON

amilkhael commented on model/reservationsBO.php:
```diff
@@ -0,0 +1,98 @@
+    }
+    else{
+        $databaseConected->desconectar();
+        return "Error: " . mysqli_error($databaseConected);
```
JSON

amilkhael reviewed (changes requested):
Te tengo que enseñar a probar las cosas jejeje

itzel-aguirre committed:
Cambios en funciones

itzel-aguirre committed:
Reporte Excel, Conectar mesas con back y Reservaciòn

1. Agregar librerìa de Excel
2. Generar reporte y descarga de las reservaciones.
3. Conectar el formulario de las mesas con las personas y la base de
datos.
4. Hacer el controlador para generar una reservaciòn.

itzel-aguirre committed:
Prueba notepad

itzel-aguirre committed:
Merge remote-tracking branch 'upstream/master' into feature/reservation

itzel-aguirre committed:
id form

itzel-aguirre committed:
Revisar funcionamiento:
1.- Generacion Excel
2.- Listado mesas por personas
3.- Validaciones
4.- Registro de una reservación

itzel-aguirre committed:
Eliminación de codigo comentado
